Source/Particles/Collision/BinaryCollision/ParticleCreationFunc.H

```diff
@@ -204,14 +204,12 @@ public:
 
                 // If the colliding particle weight decreases to zero, remove particle by
                 // setting its id to -1
-                constexpr amrex::Long minus_one_long = -1;
                 if (w1[p_pair_indices_1[i]] <= amrex::ParticleReal(0.))
                 {
-                    particle_ptr_1[p_pair_indices_1[i]].atomicSetID(minus_one_long);
+                    amrex::Gpu::Atomic::Exch(&particle_ptr_1->id(), -1);
```

Comment:
If there's an atomic store available rather than exchange, that would be preferable, since it would avoid loading the ID as part of the atomic transaction, just to throw it away.

Reply:
Yep, we thought so too but @WeiqunZhang and I are not aware of one right away.

(The old impl. also used atomic exchange to implement `atomicSetID`.)

Reply:
For SYCL, we could create a sycl::atomic_ref and then call load. For CUDA, it's not clear what's best way to do it. We could do something like https://github.com/NVIDIA/cub/blob/9ebcf693eedd90919cded7161a1560f8ad8b352d/cub/detail/strong_store.cuh#L47, but it would be hard to maintain. Maybe we could use https://nvidia.github.io/libcudacxx/extended_api.html in the future.